File: ios/EndlessRumination/Views/TakeCardView.swift
import SwiftUI

struct TakeCardView: View {
    let take: Take
    @Environment(AppState.self) private var appState
    @State private var showReportConfirmation = false
    @State private var isReported = false

    private var display: (name: String, emoji: String, color: Color, bgColor: Color) {
        Lens.displayInfo(at: take.lensIndex)
    }

    var body: some View {
        if isReported {
            // Replace flagged take with acknowledgment + crisis resources
            reportedPlaceholder
        } else {
            takeCard
        }
    }

    private var takeCard: some View {
        VStack(alignment: .leading, spacing: 16) {
            // Lens badge + Pack badge + Flag
            HStack(spacing: 8) {
                // Voice name badge
                HStack(spacing: 8) {
                    Text(display.emoji)
                        .font(.system(size: 12))
                    Text(display.name.uppercased())
                        .font(.system(size: 12, weight: .bold))
                        .tracking(2)
                        .foregroundStyle(display.color)
                }
                .padding(.horizontal, 14)
                .padding(.vertical, 6)
                .background(display.bgColor)
                .clipShape(Capsule())

                // Pack name badge (for voice pack takes)
                if let packName = take.packName {
                    Text(packName.uppercased())
                        .font(.system(size: 9, weight: .bold))
                        .tracking(1.5)
                        .foregroundStyle(display.color.opacity(0.7))
                        .padding(.horizontal, 8)
                        .padding(.vertical, 4)
                        .background(display.color.opacity(0.08))
                        .clipShape(Capsule())
                }

                Spacer()

                // Report/flag button
                Button {
                    showReportConfirmation = true
                } label: {
                    Image(systemName: "flag")
                        .font(.system(size: 12))
                        .foregroundStyle(ERColors.dimText)
                }
            }
            .alert("Report Content", isPresented: $showReportConfirmation) {
                Button("Report & Hide", role: .destructive) {
                    withAnimation {
                        isReported = true
                    }
                }
                Button("Cancel", role: .cancel) {}
            } message: {
                Text("Flag this AI-generated perspective as inappropriate or harmful? It will be hidden.")
            }

            // Headline
            Text(take.headline)
                .font(ERTypography.serifHeadline())
                .foregroundStyle(ERColors.primaryText)
                .lineSpacing(4)

            // Body
            Text(take.body)
                .font(ERTypography.body)
                .foregroundStyle(ERColors.secondaryText)
                .lineSpacing(6)
                .fontWeight(.light)
        }
    }

    private var reportedPlaceholder: some View {
        VStack(spacing: 16) {
            Image(systemName: "flag.fill")
                .font(.system(size: 24))
                .foregroundStyle(ERColors.dimText)

            Text("This perspective has been hidden.")
                .font(.system(size: 14))
                .foregroundStyle(ERColors.secondaryText)

            Text("If you're struggling, please reach out:")
                .font(.system(size: 12))
                .foregroundStyle(ERColors.dimText)

            VStack(spacing: 10) {
                ForEach(SafetyService.crisisResources, id: \.name) { resource in
                    HStack(spacing: 8) {
                        Image(systemName: resource.action == "call" ? "phone.fill" : "message.fill")
                            .font(.system(size: 12))
                            .foregroundStyle(ERColors.accentCool)
                        Text("\(resource.name): \(resource.value)")
                            .font(.system(size: 12, weight: .medium))
                            .foregroundStyle(ERColors.primaryText)
                    }
                }
            }
        }
        .frame(maxWidth: .infinity)
        .padding(.vertical, 40)
    }
}

File: android/app/src/main/kotlin/com/endlessrumination/ui/TakeCardView.kt
package com.endlessrumination.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.endlessrumination.model.Lens
import com.endlessrumination.model.Take
import com.endlessrumination.theme.ERColors
import com.endlessrumination.theme.ERTypography

@Composable
fun TakeCardView(take: Take) {
    val display = Lens.displayInfo(take.lensIndex)

    Column(modifier = Modifier.fillMaxWidth()) {
        // Badge row
        Row(
            horizontalArrangement = Arrangement.spacedBy(6.dp),
            modifier = Modifier.padding(bottom = 16.dp)
        ) {
            // Voice name badge
            Row(
                modifier = Modifier
                    .background(display.bgColor, RoundedCornerShape(50))
                    .padding(horizontal = 14.dp, vertical = 6.dp),
                horizontalArrangement = Arrangement.spacedBy(4.dp)
            ) {
                Text(display.emoji, fontSize = 12.sp)
                Text(
                    display.name.uppercase(),
                    fontSize = 12.sp,
                    fontWeight = FontWeight.Bold,
                    letterSpacing = 2.sp,
                    color = display.color
                )
            }

            // "WISE" badge
            if (take.wise || take.isPackVoice) {
                Row(
                    modifier = Modifier
                        .background(ERColors.accentGold.copy(alpha = 0.12f), RoundedCornerShape(50))
                        .padding(horizontal = 10.dp, vertical = 5.dp),
                    horizontalArrangement = Arrangement.spacedBy(4.dp)
                ) {
                    Text("\u2728", fontSize = 9.sp)
                    Text(
                        "WISE",
                        fontSize = 10.sp,
                        fontWeight = FontWeight.Bold,
                        letterSpacing = 1.5.sp,
                        color = ERColors.accentGold
                    )
                }
            }

            // Pack name badge
            val packName = take.packName
            if (packName != null) {
                Text(
                    packName.uppercase(),
                    fontSize = 9.sp,
                    fontWeight = FontWeight.Bold,
                    letterSpacing = 1.5.sp,
                    color = display.color.copy(alpha = 0.7f),
                    modifier = Modifier
                        .background(display.color.copy(alpha = 0.08f), RoundedCornerShape(50))
                        .padding(horizontal = 8.dp, vertical = 4.dp)
                )
            }
        }

        // Headline
        Text(
            text = take.headline,
            style = ERTypography.headline.copy(color = ERColors.primaryText, lineHeight = 28.sp),
            modifier = Modifier.padding(bottom = 16.dp)
        )

        // Body (scrollable)
        Column(
            modifier = Modifier
                .weight(1f)
                .verticalScroll(rememberScrollState())
        ) {
            Text(
                text = take.body,
                style = ERTypography.body.copy(color = ERColors.secondaryText, lineHeight = 20.sp)
            )

            if (!take.wise) {
                Spacer(modifier = Modifier.height(16.dp))
                Text(
                    text = "Quick take \u00B7 Powered by Haiku",
                    fontSize = 10.sp,
                    color = ERColors.dimText
                )
            }
        }
    }
}
